src/Functions/FunctionsStringSearch.h

```diff
@@ -58,14 +60,31 @@ namespace ErrorCodes
 {
     extern const int ILLEGAL_COLUMN;
     extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+    extern const int LOGICAL_ERROR;
     extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
 }
 
-template <typename Impl>
+enum class ExecutionErrorPolicy
+{
+    Null,
+    Throw
+};
+
+template <typename Impl, ExecutionErrorPolicy execution_error_policy = ExecutionErrorPolicy::Throw>
 class FunctionsStringSearch : public IFunction
 {
 public:
-    static constexpr auto name = Impl::name;
+    inline static auto name = std::invoke(
```

Comment:
Let's avoid such monstrosities 😄 A better alternative is to simply add another `struct NameHasTokenOrNull` to "hasToken.cpp" which is passed to `DB::FunctionsStringSearch<...>`. (same for the case-insensitive variant)